### PR TITLE
Added additional documentation on containers

### PR DESCRIPTION
**Pull request by pffijt**
**This pull-request includes these changes:**

- Added pages for the documentation website
- Enhancement of azurite, kafka and redis documentation.

For Kafka and Redis it doesn't include extensive documentation on the usage of the container, but it will be at least mentioned on the documentation website.

### CODE REVIEW

**Comment by pffijt** [comment]
#256 Skipped Nginx since it is deprecated.

File: testcontainers/kafka.py [hunk 1]
```diff
@@ -11,6 +11,16 @@
 
 
 class KafkaContainer(DockerContainer):
+    """
+    Kafka container.
+
+    Example
+    -------
+    ::
```
Comment (tillahoffmann) [inline testcontainers/kafka.py:19]: Would you be up for updating all the examples to use the `doctest` directive (see [here](https://github.com/testcontainers/testcontainers-python/pull/251/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fL47-R47) for an example)? That will ensure the documentation always stays in sync with the code.

**Review by tillahoffmann — COMMENTED** [review]
Nice, thank you for extending the documentation. One follow-up comment.

File: testcontainers/kafka.py (hunk 1 above)
Comment (pffijt) [reply testcontainers/kafka.py:19]: Wow, I didn't know the doctest directive, it is amazing. :D
I am going to implement this at every test!

Comment (pffijt) [reply testcontainers/kafka.py]: @tillahoffmann added the doctest directives :)

Comment (pffijt) [reply testcontainers/kafka.py]: I changed also the sphinx tests to a seperate job in the workflow.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -64,8 +96,6 @@ jobs:
           docker run --rm -v /var/run/docker.sock:/var/run/docker.sock --network=host testcontainers-python python diagnostics.py
       - name: Build documentation
```
Comment (tillahoffmann) [inline .github/workflows/main.yml:97]: Shall we also move this one up to the `sphinx` job?

Comment (pffijt) [reply .github/workflows/main.yml:97]: Agreed

Comment (pffijt) [reply .github/workflows/main.yml]: resolved

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -6,6 +6,38 @@ on:
...
+  sphinx:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+    runs-on: ubuntu-latest
```
Comment (dennisrei) [inline .github/workflows/main.yml:18]: Why not run on a lighter linux distro?

Comment (pffijt) [reply .github/workflows/main.yml:18]: This is what GitHub makes available: https://github.com/actions/runner-images 
There are not really other linux distros available unless you self-host it, which is not needed.

File: testcontainers/neo4j.py [hunk 4]
```diff
@@ -25,13 +25,15 @@ class Neo4jContainer(DbContainer):
...
-        with Neo4jContainer() as neo4j:
-            with neo4j.get_driver() as driver:
-                with driver.session() as session:
-                    result = session.run("MATCH (n) RETURN n LIMIT 1")
-                    record = result.single()
+    .. doctest::
 
+        >>> from testcontainers.neo4j import Neo4jContainer
+
+        >>> with Neo4jContainer() as neo4j:
```
Comment (tillahoffmann) [inline testcontainers/neo4j.py:32]: Could use `with ..., ..., ...` here to keep the indents under control.

File: testcontainers/rabbitmq.py [hunk 5]
```diff
@@ -8,20 +8,19 @@
 
 class RabbitMqContainer(DockerContainer):
     """
-    Test container for RabbitMQ.
+    Test container for RabbitMQ. The example below spins up a RabbitMQ broker and uses the
+    `pika` client library (https://pypi.org/project/pika/) establish a connection to the broker.
```
Comment (tillahoffmann) [inline testcontainers/rabbitmq.py:12]:
```suggestion
    `pika` client library (https://pypi.org/project/pika/) to establish a connection to the broker.
```

File: testcontainers/azurite.py [hunk 6]
```diff
@@ -19,18 +19,24 @@
 
 class AzuriteContainer(DockerContainer):
     """
-        Azurite container.
+        The example below spins up an Azurite container and
+        shows an example to create a Blob service client with the container. The method
+        get_connection_string can be used to create a client for Blob service, Queue service
```
Comment (tillahoffmann) [inline testcontainers/azurite.py:24]:
```suggestion
        :code:`get_connection_string` can be used to create a client for Blob service, Queue service
```

File: testcontainers/neo4j.py (hunk 4 above)
Comment (pffijt) [reply testcontainers/neo4j.py]: resolved

File: testcontainers/neo4j.py [hunk 7]
```diff
@@ -25,13 +25,15 @@ class Neo4jContainer(DbContainer):
...
-                with driver.session() as session:
-                    result = session.run("MATCH (n) RETURN n LIMIT 1")
-                    record = result.single()
+    .. doctest::
 
+        >>> from testcontainers.neo4j import Neo4jContainer
+
+        >>> with Neo4jContainer() as neo4j, \
+                neo4j.get_driver() as driver, \
+                driver.session() as session:
```
Comment (tillahoffmann) [inline testcontainers/neo4j.py:34]:
```suggestion
                    neo4j.get_driver() as driver, \
                    driver.session() as session:
```

**Review by tillahoffmann — APPROVED** [review]

**Comment by tillahoffmann** [comment]
This is great, thank you!

**Pull request merged**